docs/dynamicproxy-introduction.md

```diff
@@ -47,8 +47,9 @@ Schematic view of how DynamicProxy works.
 The picture above shows schematically how that works.
 
 * The blue rectangle is the proxy. Someone calls a method on the proxy (denoted by yellow arrow). Before the method reaches the target object it goes through a pipeline of interceptors.
-* Each interceptor gets a `IInvocation` object (which is another important interface from Dynamic Proxy), that holds all the information about current request, like the `MethodInfo` of the method called, along with its parameters and returned value, reference to the proxy, as well as proxied object, and few others. Each interceptor gets its chance to inspect and change those values before the actual method on the target object is called. So for example at this stage you can log debug information about what parameters were passed to the method, or validate them. Then, the interceptor has to call `invocation.Proceed()`, to pass control further down the pipeline. An interceptor can call `Proceed` at most once, otherwise an exception is thrown.
-* After last interceptor calls `Proceed`, the actual method on proxied object is invoked, and then the call travels back, up the pipeline (green arrow) giving each interceptor chance to inspect and act on, returned value, or thrown exceptions.
+* Each interceptor gets an `IInvocation` object (which is another important interface from DynamicProxy) that holds all the information about the current request, such as the `MethodInfo` of the intercepted method, along with its parameters and preliminary return value; references to the proxy and the proxied object; and a few other bits. Each invoked interceptor gets a chance to inspect and change those values before the actual method on the target object is called. For example, an interceptor may log debug information about the arguments passed to the method, or validate them.
+* Each interceptor can call `invocation.Proceed()` to pass control further down the pipeline. Interceptors usually call this method just once, but multiple calls are allowed, e.g. when implementing a "retry". (It is also permissible to cut short the interception pipeline and omit the call to `Proceed` altogether.)
+* When the last interceptor calls `Proceed`, the actual method on the proxied object is invoked, and then the call travels back, up the pipeline (green arrow) giving each interceptor another chance to inspect and act on the returned value or thrown exceptions.
```

Comment:
Here's a small side-by-side summary of the relevant changes (typo, grammar, and punctuation fixes are not marked):

![image](https://user-images.githubusercontent.com/104481/55267638-b4dbae00-5283-11e9-99bf-21b1577ebf0a.png)